src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -545,6 +545,33 @@ def prepare_latents(self, batch_size, num_channels_latents, height, width, dtype
         latents = latents * self.scheduler.init_noise_sigma
         return latents
 
+    def set_clip_output_layer(self, output_layer_idx: int = 12):
+        """Sets layer index for CLIP to use during `encode_prompt()`. Useful for experimenting with techniques like CLIP-skip.
+
+        Args:
+            output_layer_idx (`int`, *optional*, defaults to 12):
+                The output layer to use for computing the prompt embeddings.
+        """
+        if not hasattr(self, "text_encoder"):
+            raise ValueError("Cannot use `set_clip_output_layer` without a `text_encoder`.")
+
+        if not hasattr(self, "text_encoder_output_layer_idx"):
+            self.text_encoder_output_layer_idx = None
+
+        dtype, device = self.text_encoder.dtype, self.text_encoder.device
+
+        self.text_encoder = self.text_encoder.__class__.from_pretrained(
```

Comment:
I think we should actually rewrite the `encode_prompt` a bit to use `output_hidden_states=...` so that we can then simply pick the correct layer from the hidden states. The problem with the current implementation is that:
- it's quite slow
- we expect that `self.text_encoder.config._name_or_path` always works which is not always the case

Could we instead add `output_hidden_states=True` here: https://github.com/huggingface/diffusers/blob/6c314ad0ce8f59f24a26556488972bb70bd904a8/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L283